### ISSUE TITLE
about isocodes

### ISSUE THREAD

**Issue by aborrero** [rpm-software-management/rpmlint#72]
Hi there!

It seems rpmlint tries to download iso-codes at build time (tools/generate-isocodes.py).

In debian, we don't allow packages to have network traffic during build. I've solved this by pointing to static files as deployed by the iso-codes debian package. (BTW, the URLs seems down).

Do you use iso-codes inside rpmlint or just for tests or something? Isn't better to dinamically read iso-codes at runtime rather than at build time?

best regards.

**Comment by scop** [comment]
`__isocodes__.py` is generated during build over the network, but only if it's older than `tools/generate-isocodes.py` or missing, see `Makefile`. Neither should be true under normal circumstances.

grepping for `LANGUAGES` shows what we use the data for.

Looks like the iso-codes project has dropped the XML files and will in the future only ship JSON ones. With changes like this possible in the project, I think it's especially good that we pregenerate the list we need, instead of exposing rpmlint users to the breakage.

**Comment by scop** [comment]
BTW you can use the environment variables `ISO_3166_URL`, `ISO_639_URL`, and `ISO_639_3_URL` to point to local files if you want to regenerate `__isocodes__.py` during build using them.

**Comment by aborrero** [comment]
On 21 April 2016 at 17:06, Ville Skyttä <notifications@github.com> wrote:

> __isocodes__.py is generated during build over the network, but only if
> it's older than tools/generate-isocodes.py or missing, see Makefile.
> Neither should be true under normal circumstances.
>

Well, I found the issue because a 404 against the debian links, so things
happens.


> grepping for LANGUAGES shows what we use the data for.
>
> Looks like the iso-codes project has dropped the XML files and will in the
> future only ship JSON ones. With changes like this possible in the project,
> I think it's especially good that we pregenerate the list we need, instead
> of exposing rpmlint users to the breakage.
>

They do ship the XMLs, but in the installed package, not via the URL you
are using.


-- 
Arturo Borrero González

**Comment by aborrero** [comment]
On 21 April 2016 at 17:12, Ville Skyttä <notifications@github.com> wrote:

> BTW you can use the environment variables ISO_3166_URL, ISO_639_URL, and
> ISO_639_3_URL to point to local files if you want to regenerate
> __isocodes__.py during build using them.
>

Indeed, see this
https://github.com/aborrero/pkg-rpmlint/blob/master/debian/rules

-- 
Arturo Borrero González

**Comment by scop** [comment]
http://anonscm.debian.org/cgit/pkg-isocodes/iso-codes.git/tree/Makefile.am#n6
"For this release, just install the JSON files."

**Comment by aborrero** [comment]
On 21 April 2016 at 17:35, Ville Skyttä <notifications@github.com> wrote:

>
> http://anonscm.debian.org/cgit/pkg-isocodes/iso-codes.git/tree/Makefile.am#n6
> "For this release, just install the JSON files."
>
Hi Ville,

The code you are pointing to was added several releases ago (check the git
log). I've tested the package today, with the latest version in the Debian
archive and all was fine.

The XML files are there. In understand the comment could be misleading.
-- 
Arturo Borrero González
